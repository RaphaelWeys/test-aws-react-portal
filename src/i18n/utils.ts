import i18n from 'i18next';

export const i18nRfoState = (value: string): string => i18n.t(`rfostate-${value}`);
