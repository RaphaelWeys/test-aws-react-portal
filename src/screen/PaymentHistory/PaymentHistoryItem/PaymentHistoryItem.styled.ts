import styled, { css } from 'styled-components';

export const Icon = styled.div`
  margin-right: 30px;
`;

export const Date = styled.div`
  margin-right: 70px;
`;

export const Title = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: flex-start;

  > :first-child {
    font-weight: bold;
    margin-bottom: 9px;
  }

  > :nth-child(2) {
    font-size: 13px;
  }
`;

export const Center = styled.div`
  display: flex;
  flex-direction: column;
  align-items: flex-start;

  ${({ theme: { breakpoints } }) => css`
    @media (min-width: ${breakpoints.md}) {
      align-items: center;
      flex-direction: row;
    }
  `}
`;

export const CenterPrice = styled.div`
  display: flex;
  align-items: center;
  margin-top: 20px;
  align-self: flex-end;

  ${({ theme: { breakpoints } }) => css`
    @media (min-width: ${breakpoints.sm}) {
      margin-top: 0;
      align-self: auto;
    }
  `}
`;
