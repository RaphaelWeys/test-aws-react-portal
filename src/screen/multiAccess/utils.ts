export const hasSomeClientMarketActivated = (clientMarkets) => clientMarkets?.some((market) => market.enabled);
