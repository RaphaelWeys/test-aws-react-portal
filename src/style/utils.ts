import styled, { css } from 'styled-components';

type IJustify = 'flex-start' | 'flex-end' | 'center' | 'space-between' | 'space-around' | 'space-evenly';
type IAlign = 'stretch' | 'center' | 'flex-start' | 'flex-end' | 'baseline' | 'initial' | 'inherit';
type IDirection = 'row' | 'row-reverse' | 'column' | 'column-reverse';
type IDisplay = 'inline-block';

export const AlignItems = styled.div<{
  justify?: IJustify;
  align?: IAlign;
  flex?: boolean;
  noFlexWrap?: boolean;
  space?: number;
  direction?: IDirection;
  bothSideSpace?: boolean;
  display?: IDisplay;
  allSpace?: boolean;
}>`
  ${({
    justify = 'flex-start',
    align = 'center',
    flex = false,
    noFlexWrap = false,
    space = 0,
    direction = 'row',
    bothSideSpace,
    display,
    allSpace,
  }) => css`
    align-items: ${align};
    display: ${display || 'flex'};
    justify-content: ${justify};
    flex-direction: ${direction};
    ${!noFlexWrap && 'flex-wrap: wrap;'};

    > * {
      ${flex && 'flex: 1'};

      ${bothSideSpace
        ? space && (direction === 'row' || direction === 'row-reverse')
          ? `margin-right: ${space / 16}rem; margin-left: ${space / 16}rem`
          : `margin-top: ${space / 16}rem; margin-bottom: ${space / 16}rem`
        : null};

      ${allSpace ? `margin: ${space / 16}rem` : null}
    }
    > * + * {
      ${space && (direction === 'row' || direction === 'row-reverse')
        ? bothSideSpace
          ? `margin-right: ${space / 16}rem; margin-left: ${space / 16}rem`
          : `margin-left: ${space / 16}rem`
        : bothSideSpace
        ? `margin: ${space / 16}rem 0`
        : `margin-top: ${space / 16}rem`};
    }
  `}
`;

export const Margin = styled.div<{ mt?: number; mb?: number; ml?: number; mr?: number }>`
  ${({ mt, mb, ml, mr }) => css`
    ${mt && `margin-top: ${mt}px`};
    ${mb && `margin-bottom: ${mb}px`};
    ${ml && `margin-left: ${ml}px`};
    ${mr && `margin-right: ${mr}px`};
  `}
`;

export const DisplayErrorType = styled.div`
  color: red;
  font-size: 0.75rem;
  min-height: 12px;
  width: 100%;
  margin-top: 8px;
  display: flex;
  align-items: center;
`;

export const Error = styled.span`
  ${({ theme: { colors, fontSize } }) => css`
    font-size: ${fontSize.extraSmall};
    color: ${colors.red};
  `}
`;

export const BoldBlack = styled.div`
  color: #3c3c3c;
  font-weight: bold;
`;

export const HeaderOne = styled.div`
  ${({ theme: { colors, fontSize } }) => css`
    font-size: ${fontSize.header};
    color: ${colors.black};
    font-family: 'Fira Sans', serif;
    font-weight: bold;
  `}
`;

export const HeaderThree = styled.div`
  ${({ theme: { colors, fontSize } }) => css`
    font-size: ${fontSize.smallHeader};
    color: ${colors.black};
    font-family: 'Fira Sans', serif;
    font-weight: 600;
  `}
`;

export const TextRegular = styled.div`
  ${({ theme: { colors, fontSize } }) => css`
    font-size: ${fontSize.regular};
    color: ${colors.grayDark};
  `}
`;

export const TextRegularBold = styled.div`
  ${({ theme: { colors, fontSize } }) => css`
    font-size: ${fontSize.regular};
    color: ${colors.grayDark};
    font-weight: bold;
  `}
`;

export const Label = styled.label`
  ${({ theme: { colors, fontSize } }) => css`
    font-size: ${fontSize.regular};
    color: ${colors.black};
    font-weight: 600;
  `}
`;

export const BlueStyle = styled.div<{ disabled?: boolean }>`
  ${({ theme: { colors, fontSize }, disabled }) => css`
    font-size: ${fontSize.small};
    color: ${disabled ? colors.gray : colors.baseColor};
    font-weight: bold;
  `}
`;

export const WarningText = styled.div`
  ${({ theme: { colors, fontSize } }) => css`
    font-size: ${fontSize.small};
    color: ${colors.orange};
    font-weight: bold;
  `}
`;
