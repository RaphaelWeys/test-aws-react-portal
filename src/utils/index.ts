import Cookies from 'js-cookie';

import { TOKEN_NAME } from '../hooks/useCookie';

export const getCorrectBackend = (backend: string) => {
  switch (backend) {
    case 'follow': {
      return process.env.REACT_APP_BACKEND_FOLLOW_URL;
    }
    case 'tender': {
      return process.env.REACT_APP_BACKEND_TENDER_URL;
    }
    default: {
      return process.env.REACT_APP_BACKEND_PORTAL_URL;
    }
  }
};

export const callAll = (...fns: (((...args: any[]) => void) | undefined)[]) => (...args: any[]) =>
  fns.forEach((fn) => fn && fn(...args));

export const saveTokenCookies = (token, domain) => {
  const month13Later = new Date();
  month13Later.setMonth(month13Later.getMonth() + 13);

  Cookies.set(TOKEN_NAME, token, {
    path: '/',
    domain,
    expires: month13Later,
  });
};
