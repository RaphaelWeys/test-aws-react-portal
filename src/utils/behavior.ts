import i18n from 'i18next';

import { Item } from '../components/Select/Select';
import { COUNTRY_CONFIG } from '../config/app-config';

const i18nCountry = (value: string) => i18n.t(`country-${value}`);

export const i18nItemsForCountry = (): Item[] =>
  COUNTRY_CONFIG.countries.map((country) => ({
    label: i18nCountry(country),
    value: country,
    key: country,
  }));

export const formatStringToBoolean = <T extends { [key: string]: any }>(obj: T): T =>
  Object.keys(obj).reduce((newObj, key) => {
    const value = obj[key];
    return {
      ...newObj,
      [key]: value === 'true' ? true : value === 'false' ? false : value,
    };
  }, {} as T);

export const formatBooleanToString = (obj: { [k: string]: any }) =>
  Object.keys(obj).reduce((newObj, key) => {
    const value = obj[key];
    return {
      ...newObj,
      [key]: value === true ? 'true' : value === false ? 'false' : value,
    };
  }, {});

export const isBoolean = (value: any): boolean => typeof value === 'boolean';

export const isKam = (userInfo) => userInfo.role === 'kam';
export const isManager = (userInfo) => userInfo.role === 'manager';
export const isClient = (userInfo) => userInfo.role === 'client';
export const isMultiAccess = (userInfo) => userInfo.isMultiAccess;
