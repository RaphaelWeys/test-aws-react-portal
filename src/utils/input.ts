import i18n from 'i18next';
import { FieldError, Message } from 'react-hook-form';

export const getErrorMessage = (error: FieldError, type?: string): Message => {
  if (typeof error.message === 'string' && error.message.length > 0) return i18n.t(error.message);
  if (error.type === 'required') return i18n.t(`global-field-${type}-required`);
  if (error.type === 'pattern') return i18n.t(`global-field-${type}-pattern`);

  return 'Error field';
};
