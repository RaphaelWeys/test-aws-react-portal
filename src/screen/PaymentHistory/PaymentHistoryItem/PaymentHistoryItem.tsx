import React, { FC } from 'react';
import styled, { css } from 'styled-components';
import { useTranslation } from 'react-i18next';

import { Center, Date, Icon, Title, CenterPrice } from './PaymentHistoryItem.styled';
import InvisibleButton from '../../../components/InvisibleButton';
import DownloadIcon from '../../../components/icons/DownloadIcon';
import { getCurrencyFormatted } from '../../../utils/number';

interface Props {
  className?: string;
  icon: string;
  startDate: string;
  title: string;
  description: string;
  amount: number;
  urlToRedirect: string;
  isLast: boolean;
}

const PaymentHistoryItem: FC<Props> = ({ className, icon, startDate, title, description, amount, urlToRedirect }) => {
  const [t] = useTranslation();
  const handleClickDownload = () => {
    const win = window.open(urlToRedirect, '_blank');
    if (win) win.focus();
  };

  return (
    <div className={className} data-testid="payment-history-item">
      <Center>
        <Icon>
          <img src={icon} alt="picto logo" />
        </Icon>
        <Date>{startDate}</Date>
        <Title>
          <div>{title}</div>
          {description && (
            <div>
              {t('payment-history-reference')}: {description}
            </div>
          )}
        </Title>
      </Center>
      <CenterPrice>
        <div data-testid="price-test">{getCurrencyFormatted(amount)}</div>
        {urlToRedirect && (
          <InvisibleButton data-testid="button-download" onClick={handleClickDownload}>
            <DownloadIcon />
          </InvisibleButton>
        )}
      </CenterPrice>
    </div>
  );
};

export default styled(PaymentHistoryItem)`
  ${({ isLast }) => css`
    display: flex;
    justify-content: space-between;
    flex-direction: column;
    align-items: flex-start;
    padding: 16px 16px 13px;
    border-top: 2px dotted black;
    ${isLast && 'border-bottom: 2px dotted black'};

    ${({ theme: { breakpoints } }) => css`
      @media (min-width: ${breakpoints.sm}) {
        flex-direction: row;
        align-items: center;
      }
    `}
  `}

  ${InvisibleButton} {
    margin-left: 30px;
  }
`;
