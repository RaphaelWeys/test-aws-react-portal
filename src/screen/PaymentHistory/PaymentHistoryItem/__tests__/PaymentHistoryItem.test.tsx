import React from 'react';
import 'jest-styled-components';
import { render, fireEvent } from '@testing-library/react';

import PaymentHistoryItem from '../PaymentHistoryItem';
import { formatNumber } from '../../../../utils';

const getProps = (override?: { [key: string]: string | number | boolean | null }) => ({
  className: 'sc-pJurq cwzEZF',
  icon: '/static/media/picto-optimization.5dca1da9.svg',
  startDate: '01/07/2020',
  title: 'YEM Optimization subscription',
  description: '1 year subscription - 4 contracts',
  amount: 1800,
  urlToRedirect:
    'https://pay.stripe.com/receipts/acct_1G877uAkgNStjmVM/ch_1H03X4AkgNStjmVMZ9ZxDFqL/rcpt_HZBusReohafFkqBA44tWB8CjHgRoQZX',
  isLast: true,
  ...override,
});

test('it should open new tab with the invoice', () => {
  const props = getProps();
  const { getByTestId } = render(<PaymentHistoryItem {...props} />);

  const { open } = window;
  delete window.open;
  // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
  // @ts-ignore
  window.open = jest.fn();

  fireEvent.click(getByTestId(/button-download/i));

  expect(window.open).toHaveBeenCalledWith(props.urlToRedirect, '_blank');
  window.open = open;
});

test('It should match snapshot', async () => {
  const { container } = render(<PaymentHistoryItem {...getProps()} />);

  expect(container).toMatchSnapshot();
});

test('It should display the price with the correct format', () => {
  const props = getProps();
  const { getByTestId } = render(<PaymentHistoryItem {...props} />);

  expect(getByTestId(/price-test/i)).toHaveTextContent(`${formatNumber(props.amount)} â‚¬`);
});

test('It should hide the invoice button if amount is 0', () => {
  const props = getProps({ urlToRedirect: null });
  const { queryByTestId } = render(<PaymentHistoryItem {...props} />);

  expect(queryByTestId(/button-download/i)).toBeNull();
});
