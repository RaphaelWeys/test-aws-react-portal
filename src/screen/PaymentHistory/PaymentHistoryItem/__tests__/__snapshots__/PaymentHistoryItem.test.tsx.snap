// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should match snapshot 1`] = `
.c2 {
  margin-right: 30px;
}

.c3 {
  margin-right: 70px;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c4 > :first-child {
  font-weight: bold;
  margin-bottom: 9px;
}

.c4 > :nth-child(2) {
  font-size: 13px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c6 {
  background-color: transparent;
  border: 0;
  outline: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0;
}

.c6:hover {
  cursor: pointer;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 16px 16px 13px;
  border-top: 2px dotted black;
  border-bottom: 2px dotted black;
}

.c0 .c5 {
  margin-left: 30px;
}

<div>
  <div
    class="c0 sc-pJurq cwzEZF"
    data-testid="payment-history-item"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <img
          alt="picto logo"
          src="/static/media/picto-optimization.5dca1da9.svg"
        />
      </div>
      <div
        class="c3"
      >
        01/07/2020
      </div>
      <div
        class="c4"
      >
        <div>
          YEM Optimization subscription
        </div>
        <div>
          payment-history-reference
          : 
          1 year subscription - 4 contracts
        </div>
      </div>
    </div>
    <div
      class="c1"
    >
      <div
        data-testid="price-test"
      >
        1 800.00
         €
      </div>
      <button
        class="c5 c6"
        data-testid="button-download"
        type="button"
      >
        <svg
          height="18.867"
          viewBox="0 0 18.492 18.867"
          width="18.492"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            id="download"
            transform="translate(0.5)"
          >
            <g
              data-name="Groupe 708"
              id="Groupe_708"
              transform="translate(17.492 18.367) rotate(180)"
            >
              <path
                d="M0,0V6.945H17.492V0"
                data-name="Tracé 85"
                fill="none"
                id="Tracé_85"
                stroke="#727577"
                stroke-miterlimit="10"
                stroke-width="1"
                transform="translate(17.492 6.945) rotate(180)"
              />
              <path
                d="M15.023,0,7.511,7.512,0,0"
                data-name="Tracé 86"
                fill="none"
                id="Tracé_86"
                stroke="#727577"
                stroke-miterlimit="10"
                stroke-width="1"
                transform="translate(16.258 13.017) rotate(180)"
              />
              <line
                data-name="Ligne 33"
                fill="none"
                id="Ligne_33"
                stroke="#727577"
                stroke-miterlimit="10"
                stroke-width="1"
                transform="translate(8.746 18.367) rotate(180)"
                y1="12.862"
              />
            </g>
          </g>
        </svg>
      </button>
    </div>
  </div>
</div>
`;
